scheduler/src/scheduling.ts

```diff
@@ -3,217 +3,179 @@ import deepEqual from 'deep-equal'
-        }
-      } else {
-        console.debug(`Triggering datasource ${datasourceId} failed:`, httpError.message)
-      }
-      if (i === MAX_TRIGGER_RETRIES - 1) { // last retry
-        console.error(`Could not trigger datasource ${datasourceId}:`, httpError)
+  async execute (datasourceConfig: DatasourceConfig): Promise<void> {
+    const datasourceId = datasourceConfig.id
+    for (let i = 0; i < this.triggerRetries; i++) {
+      try {
+        await AdapterClient.triggerDatasource(datasourceId)
+        console.log(`Datasource ${datasourceId} triggered.`)
         break
+      } catch (error) {
+        if (isAxiosError(error)) {
+          handleAxiosError(error)
+        }
+        if (i === this.triggerRetries - 1) { // last retry
+          console.error(`Could not trigger datasource ${datasourceId}`)
+          break
+        }
+        console.info(`Triggering datasource failed - retrying (${i}/${this.triggerRetries})`)
       }
-      console.info(`Triggering datasource failed - retrying (${i}/${MAX_TRIGGER_RETRIES})`)
     }
+    this.reschedule(datasourceConfig)
   }
-  reschedule(datasourceConfig)
-}
 
-function isAxiosError (error: any): error is AxiosError {
-  return error.isAxiosError
-}
+  async upsertJob (datasourceConfig: DatasourceConfig): Promise<ExecutionJob> {
+    const isNewDatasource = !this.existsJob(datasourceConfig.id)
+    const datasourceState = isNewDatasource ? 'New' : 'Updated'
 
-export async function upsertJob (datasourceConfig: DatasourceConfig): Promise<ExecutionJob> {
-  const isNewDatasource = !existsJob(datasourceConfig.id)
-  const datasourceState = isNewDatasource ? 'New' : 'Updated'
+    console.log(`[${datasourceState}] datasource detected with id ${datasourceConfig.id}.`)
 
-  console.log(`[${datasourceState}] datasource detected with id ${datasourceConfig.id}.`)
+    if (!isNewDatasource) {
+      this.cancelJob(datasourceConfig.id)
+    }
 
-  if (!isNewDatasource) {
-    cancelJob(datasourceConfig.id)
+    return this.scheduleDatasource(datasourceConfig)
   }
 
-  return scheduleDatasource(datasourceConfig)
-}
+  getAllJobs (): ExecutionJob[] {
+    return Array.from(this.allJobs.values())
+  }
 
-export function getAllJobs (): ExecutionJob[] {
-  return Array.from(allJobs.values())
+  cancelAllJobs (): void {
+    this.allJobs.forEach(job => {
+      schedule.cancelJob(job.scheduleJob)
+    })
+    this.allJobs.clear()
+  }
+
+  cancelJob (jobId: number): void {
+    const job = this.allJobs.get(jobId)
+    job?.scheduleJob.cancel()
+  }
 }
 
-export function cancelAllJobs (): void {
-  allJobs.forEach(job => {
-    schedule.cancelJob(job.scheduleJob)
-  })
-  allJobs.clear()
+const isAxiosError = function (error: any): error is AxiosError {
+  return error.isAxiosError
 }
 
-export function cancelJob (jobId: number): void {
-  const job = allJobs.get(jobId)
-  job?.scheduleJob.cancel()
+const handleAxiosError = function (error: AxiosError): void {
```

Comment:
Also this method lacks of easy readability because of the if statements.

Here is a refactoring proposal.

```js
 const handleAxiosError = function (error: AxiosError): void {
    const baseMsg = 'Error during datasource triggering: ' 
    if (error.response !== undefined) {
      console.error(`${baseMsg} [${error.response.status}] ${error.response.data}`)
      return
    } 
    
    if (error.request !== undefined) {
        console.error(`${baseMsg} Request failed: ${JSON.stringify(error.request)}`)      
        return
    }

    console.error(`${baseMsg} unknown reason`)    
  }
```

Reply:
Why do you consider multiple `if ... return` statements more readable than `if ... else`? I don't have a strong opinion on that so I don't mind changing it, but I am curious anyway.

Reply:
IMHO the `if ... return` (also known as Guard Clauses) make the code more lean and readable.
1) The blank line between the if statements helps to identify the different outcomes more easily. Whereas the if-else statement looks like a big blob. 
2) Readability does not decrease in the way it does with if-else statements when you add more checks.
3) You avoid unnecessary branching in complex scenarios.

Reply:
Alright, thanks for the clarification!